### DISCUSSION TITLE
New joins frequently don't notice the Captcha challenge

### DISCUSSION THREAD

**Discussion by Marvin-YYC** [J-Rios/TLG_JoinCaptchaBot#199] [Q&A]
Thanks for bot.  It's great!  
It appears that some people don't notice the captcha challenge and end up getting removed from the group after the time runs out.  Is there some way to set the bot to send a second @mention challenge to the new join after a minute or two? They sometimes think they have been banned and don't try to rejoin.

**Comment by J-Rios** [comment]
Hello!

Indeed this is a well known behaviour that I have seen from the very beggining version of the Bot... Afraid I didn't find any way to ensure that people read the message, the Bot send the message with username mention (if the user has an username/alias configured) to ensure the user receive a notification about the message, however there are people that just ignore the message and notification...

I see your point, but I'm not convinced with the proposed solution, due I see some problems with sending the captcha message multiple times:

1. Making the Bot send more captcha messages after 1/2 mins for people that has not complete the captcha yet, implies to add a new process that has a high impact in required Bot resources (RAM and CPU, notice the Bot is running for thousand of peoples at the same time).

2. Making the Bot send more captcha messages after 1/2 mins for people that has not complete the captcha yet, will cause a lot of confussion in groups where there are multiple people joining at the same time.

3. If that kind of users doesn't read the first captcha message when they join the group (are more active), why should we expect that they are going to read following messages? I really think that people just ignore the first captcha message because they don't want to spend any time reading anything at that moment, they just want to join a group, and maybe come back to it in few hours, so no matter if we send more messages in that 5 mins.

So I feel in my guts that this approach is not worth it and won't solve the problem anyway.

Anyway, I will keep this discussion open if there are other ideas in the future for this problem.

**Reply by Marvin-YYC** [reply]
Thank you for responding.  😊
 It is odd that they don't see the capture.  I wonder If they are scolling through the group history, and don't notice the captcha message?!

If I happen to notice the captcha message and see that they haven't  tried to respond I will inform them that they only have a couple minutes to complete the captcha before they are removed. They usually complete the captcha successfully after my prompting. That's why I thought a second reminder might be helpful.  A few people I have chatted with thought they had been banned as they hadn't seen the captcha message at all. 

One other thought, what if the captcha message was sent 60 seconds after the person joined?  Perhaps this would increase the likelyhood that they would see it?

I know other people that are also using your bot, and are having the same problem.  Seems to be about 30% of new joins don't see the captcha.   50% of them don't try to join again. I can only think that they are new to telegram,  and are not expecting a captcha message. Or they are bots and your bot service is working perfectly. 👍

**Reply by Marvin-YYC** [reply]
I had someone try to join a couple of my groups yesterday and didn't respond to the captcha.   
Today they tried again and were successful.   They told me that yesterday they were using the windows telegram desktop app and the captcha didn't appear.   

Today they used the iOS app and the captcha worked fine.  

This is not the first time I've heard of this same problem with the desktop app.
 ¯\_(ツ)_/¯

**Reply by J-Rios** [reply]
Interesting, however that issue is not related to the Bot. Notice that the Bot only speak with Telegram Server, then the Telegram Server is who send the messages to all the Telegram clients, so if the messages doesn't appear in a Telegram client Application, then the issue/bug is in that Telegram Application (or the user network).

Personally I didnt face any issue with Telegram Desktop app client in both Windows and Linux, I used It everyday, but of course I use the official Application and not any unofficial app that could exists.

**Comment by Marvin-YYC** [comment]
I had a couple people try to sign into two of my groups today, and they said the captcha message wasn't very obvious?. . .   They also said that when they joined the groups they were automatically sent to the oldest message in the default topic and the captcha was at the very bottom, newest message.  I tried signing into a couple of my own groups again, with the windows app, and I didn't see that at all.  
No idea . . . perhaps people are just not paying attention or they need to update their apps.   Thanks anyway :)
